compiler/rustc_session/src/cstore.rs

```diff
@@ -68,6 +68,9 @@ pub enum LinkagePreference {
 pub struct NativeLib {
     pub kind: NativeLibKind,
     pub name: Option<Symbol>,
+    /// With packed_bundled_libs enabled, path to library is stored in metadata.
+    /// It's empty otherwise.
+    pub path: PathBuf,
```

Comment:
Can this be an `Option`?

Reply:
Also putting the path in the metadata would break reproducibility. Would using ClearCrossCrate work?

Reply:
>Would using ClearCrossCrate work?

Probably not, the `file_name` component of the path is still needed cross crate.

Reply:
I don't want to spend too much of @BelovDV's time on figuring out the metadata encoding details.
I guess the simplest solution is to keep only the `file_name` component in metadata and then in `fn link_rlib` re-find the full path again using `fn find_native_static_library` in verbatim mode.